### PR TITLE
Update Remote API support to v1.16

### PR DESCRIPTION
**Pull request by tduffield**
- Update API version to 1.16
- Add Docker::Exec#json
- Add exitstatus to array returned by Docker::Exec#start
- Add check for new Docker#info parameters
- Remove Docker::Exec#resize tests because they were not working reliably.

### CODE REVIEW

**Comment by tduffield** [comment]
@tlunter fixed your note about documentation and rebased on the latest master.

**Comment by tlunter** [comment]
Looks like we're still running into an error on travis. Any ideas?

**Comment by tduffield** [comment]
Looks like the errors are occurring because of the latest PR. I'm currently trying to work through the solution.

**Comment by carlossg** [comment]
+1 I was looking for the exec ExitCode too. 

In exec by just returning an array it is getting limited, only ExitCode is returned from all the data docker exposes in inspect, and it doesn't return the ordered output (with both stdout and stderr)

Should it return a hash with all the data? it would be a breaking change though

File: lib/docker/exec.rb [hunk 1]
```diff
@@ -35,7 +41,7 @@ def self.create(options = {}, conn = Docker.connection)
   # @option options [TrueClass, FalseClass] :tty (false) Whether to attach using
   #     a pseudo-TTY.
   #
-  # @return [Array, Array] The STDOUT and STDERR responses
+  # @return [Array, Array, String] The STDOUT, STDERR and exit code
```
Comment (calebthompson) [inline lib/docker/exec.rb]: The README shows an integer exit code, but your doc change shows String.

**Comment by tlunter** [comment]
Release in v1.19.0

**Pull request closed without merging**